scripts/setup-new-module

```diff
@@ -0,0 +1,352 @@
+        <groupId>software.amazon.awssdk</groupId>
+        <artifactId>aws-sdk-java-pom</artifactId>
+        <version>${SDK_VERSION}</version>
+    </parent>
+
+    <artifactId>${MODULE_NAME}</artifactId>
+    <name>AWS Java SDK :: ${MODULE_NAME}</name>
+    <description>AWS SDK for Java - ${MODULE_NAME}</description>
+    <url>https://aws.amazon.com/sdkforjava</url>
+EOF
+
+# Add Automatic-Module-Name for non-test modules
+if [ "$IS_TEST_MODULE" = false ]; then
+  cat >> "$MODULE_DIR/pom.xml" << EOF
+    
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <manifestEntries>
+                            <!-- should be same as package name -->
+                            <Automatic-Module-Name>software.amazon.awssdk.TODO</Automatic-Module-Name>
+                        </manifestEntries>
+                    </archive>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+EOF
+fi
+
+# Close pom.xml
+cat >> "$MODULE_DIR/pom.xml" << EOF
+
+    <dependencies>
+        <!-- Add your dependencies here -->
+    </dependencies>
+
+</project>
+EOF
+
+echo "Basic module structure created at $MODULE_DIR"
+
+# Function to add dependency to a pom.xml file
+add_dependency_to_pom() {
+    local pom_file=$1
+    local module_name=$2
+    
+    # Check if the file exists
+    if [ ! -f "$pom_file" ]; then
+        echo "Warning: $pom_file does not exist. Skipping."
+        return
+    fi
+    
+    # Check if dependency already exists
+    if grep -q "<artifactId>$module_name</artifactId>" "$pom_file"; then
+        echo "Dependency already exists in $pom_file. Skipping."
+        return
+    fi
+    
+    # Find the dependencies section and add the new dependency
+    sed -i.bak "/<dependencies>/a\\
+        <dependency>\\
+            <groupId>software.amazon.awssdk</groupId>\\
+            <artifactId>$module_name</artifactId>\\
+            <version>\${awsjavasdk.version}</version>\\
+        </dependency>\\
+" "$pom_file"
+    
+    # Remove backup file
+    rm -f "${pom_file}.bak"
+    
+    echo "Added dependency to $pom_file"
+}
+
+# Function to update the root pom.xml to include the new module
+update_root_pom() {
+    local root_pom="$ROOT_DIR/pom.xml"
+    local module_name=$1
+
+    # Check if the file exists
+    if [ ! -f "$root_pom" ]; then
+        echo "Warning: $root_pom does not exist. Skipping."
+        return
+    fi
+
+    # Determine the module path based on whether it's a test module
+    local module_path="$module_name"
+    if [ "$IS_TEST_MODULE" = true ]; then
+        module_path="test/$module_name"
+    fi
+
+    # Check if module already exists
+    if grep -q "<module>$module_path</module>" "$root_pom"; then
+        echo "Module already exists in root pom.xml. Skipping."
```

Comment:
When would this happen and is this something that should be an error?

Reply:
It could happen when the user has already set up some files manually because they may not know this script exist when they created the new module